include/deal.II/base/function.templates.h

```diff
@@ -45,9 +45,11 @@ Function<dim, Number>::Function (const unsigned int n_components,
 }
 
 
+
+// The destructor is pure virtual so we can't default it
+// in the declaration.
 template <int dim, typename Number>
-Function<dim, Number>::~Function ()
-{}
+Function<dim, Number>::~Function () = default;
```

Comment:
Oh, how interesting. This doesn't work because you can't write `=0 =default`?

Reply:
Yes, exactly, see https://stackoverflow.com/questions/11494422/default-pure-virtual-destructor